### PR TITLE
update to monorepo & v1.1.0

### PR DESCRIPTION
**Pull request by vanderian**
initial push to check e2e & CI flow

### CODE REVIEW

File: pallets/bootstrap/Cargo.toml [hunk 1]
```diff
@@ -2,70 +2,75 @@
...
-version = '0.1.0'
+name = "pallet-bootstrap"
+version = "0.1.0"
 
 [package.metadata.docs.rs]
-targets = ['x86_64-unknown-linux-gnu']
+targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false }
```
Comment (vanderian) [inline pallets/bootstrap/Cargo.toml:12]: dependencies are now sorted alphabetically

File: pallets/bootstrap/Cargo.toml [hunk 2]
```diff
@@ -2,70 +2,75 @@
...
 std = [
-	'serde',
-	'codec/std',
-	'sp-std/std',
-	'sp-runtime/std',
-	'sp-io/std',
-	'frame-support/std',
-	'frame-system/std',
-	'orml-tokens/std',
+	"codec/std",
```
Comment (vanderian) [inline pallets/bootstrap/Cargo.toml:55]: sorted alphabetically as well

File: runtime/common/src/migration.rs [hunk 3]
```diff
@@ -24,60 +24,13 @@ where
...
-
-		Ok(())
+		let version = orml_asset_registry::Pallet::<T>::on_chain_storage_version();
+		if version == 2 {
+			info!(target: "asset-registry", "No migration applied, remove");
+			T::DbWeight::get().reads(1)
+		} else {
+			StorageVersion::new(2).put::<orml_asset_registry::Pallet<T>>();
+			T::DbWeight::get().reads_writes(1, 1)
+		}
```
Comment (vanderian) [inline runtime/common/src/migration.rs:34]: we are missing a correct version in asset-registry on kusama
rococo was sudo updated to '2'

File: runtime/common/src/lib.rs [hunk 4]
```diff
@@ -82,6 +74,33 @@ pub mod currency {
 	}
 }
 
+pub mod types {
```
Comment (vanderian) [inline runtime/common/src/lib.rs:77]: runtime types were removed from `mangata-types` and added here

File: runtime/common/src/lib.rs [hunk 5]
```diff
@@ -96,6 +115,28 @@ pub mod tokens {
 	}
 }
 
+/// Opaque types. These are used by the CLI to instantiate machinery that don't need to know
+/// the specifics of the runtime. They can then be made to be agnostic over specific formats
+/// of data like extrinsics, allowing for them to continue syncing the network through upgrades
+/// to even the core data structures.
+pub mod opaque {
```
Comment (vanderian) [inline runtime/common/src/lib.rs:122]: used in `node/src`

File: runtime/integration-test/src/xcm/mod.rs [hunk 6]
```diff
@@ -1,8 +1,8 @@
-#[cfg(feature = "with-kusama-runtime")]
-pub mod kusama_test_net;
+// #[cfg(feature = "with-kusama-runtime")]
+// pub mod kusama_test_net;
 
-#[cfg(feature = "with-kusama-runtime")]
-pub mod kusama_xcm_transfer;
+// #[cfg(feature = "with-kusama-runtime")]
+// pub mod kusama_xcm_transfer;
```
Comment (vanderian) [inline runtime/integration-test/src/xcm/mod.rs:5]: the old xcm-simulator is not compatible with polkadot v1
the simulator from polkadot-sdk is overly complicated, needs full genesis setup for relays, etc...should be done in separate PR

Comment (vanderian) [reply runtime/integration-test/src/xcm/mod.rs:5]: IN GENERAL, there are multiple changes involved
* upgrade to v1.1.0
* removal of types from `mangata-types` eg. `TokenId = u64`; pallets now don't use specific runtime types instead rely on traits, most of the changes are in `mock.rs` & `tests.rs` with `.into()` removals
* cleanup -> Cargo.toml is alpha sorted, all quotes unified to "
* `node/src` -> major changes, follows latest polkadot-sdk with slight modifications to suite our needs
* CI -> new builder image, try-runtime, zombienet...

File: runtime/mangata-kusama/src/lib.rs [hunk 7]
```diff
@@ -793,16 +790,20 @@ impl_runtime_apis! {
...
 			System::pop_txs(count as usize)
 		}
 
-		fn create_enqueue_txs_inherent(txs: Vec<<Block as BlockT>::Extrinsic>) -> <Block as BlockT>::Extrinsic{
+		fn create_enqueue_txs_inherent(txs: Vec<<Block as BlockT>::Extrinsic>) -> <Block as BlockT>::Extrinsic {
+			for t in txs.iter() {
+				if let Some((_, _, extra)) = &t.signature {
+					let _ = extra.additional_signed();
+				}
+			}
```
Comment (vanderian) [inline runtime/mangata-kusama/src/lib.rs:806]: this is important, a bit hackish

during block building, state access is recorded and then passed to relay to validate block
previously these 'next block' txs were validated in block builder within a transaction that was rolled back eventually, all was well
in validate function the relay validated the trxs correctly

now the rollback also reverts proof recorder
the relay in validate function now fails because of access to txs extras, specifically the mortality check with block number storage access
the storage does not exist in the supplied recorded proof

this API is called during last stages of block building, we evaluate the extras so that the storage access is recored in proof and the validation fucntion on relay succeeds

**Review by mateuszaaa — APPROVED** [review]

File: .dockerignore [hunk 8]
```diff
@@ -1,18 +1,5 @@
...
-launch
-scripts
-workflows
-Makefile
-Tiltfile
-**/target/
-**/*.txt
-**/*.md
-./docker-cargo
-./mvr/docker-cargo
```
[the remark this answers is not in the document]
Comment (vanderian) [reply .dockerignore:15]: i have removed all the individual entries and replaced with wildcard `*`
we are interested only in `mangata-node` and wasms which are filtered out with `!` entries

**Review by mateuszaaa — APPROVED** [review]
lgtm

**Review by mateuszaaa — APPROVED** [review]
lgtm
